django_mongodb_backend/query_conversion/expression_converters.py

```diff
@@ -0,0 +1,129 @@
+"""Expression To Match Converters"""
+
+
+class _BaseExpressionConverter:
+    """
+    Base class for optimizers that handle specific operations in MQL queries.
+    This class can be extended to implement optimizations for other operations.
+    """
+
+    @classmethod
+    def convert(cls, expr):
+        raise NotImplementedError("Subclasses should implement this method.")
+
+    @classmethod
+    def is_simple_value(cls, value):
+        """
+        Check if the value is a simple type (not a dict).
+        """
+        if isinstance(value, str) and value.startswith("$"):
+            return False
+        if isinstance(value, list | tuple | set):
```

Comment:
Using `|` here is interesting. Would it be better to use an explicit tuple?

Reply:
The usual way to write this is: `if isinstance(value, (list, tuple, set)):`

Reply:
I think the suggestion is `isinstance(value, (list, tuple, set)`. [It's enforced by ruff](https://docs.astral.sh/ruff/rules/non-pep604-isinstance/) to use the union operator, but I now see the rule is deprecated, so I'll look into ignoring that rule and correcting existing code.